yargs.js

```diff
@@ -1,8 +1,15 @@
 'use strict'
 
-// an async function fails early in Node.js versions prior to 8.
-async function requiresNode8OrGreater () {}
-requiresNode8OrGreater()
+// See https://github.com/yargs/yargs#supported-nodejs-versions for our
+// version support policy. The YARGS_MIN_NODE_VERSION is used for testing only.
+const minNodeVersion = (process && process.env && process.env.YARGS_MIN_NODE_VERSION)
+  ? Number(process.env.YARGS_MIN_NODE_VERSION) : 10
+if (process && process.version) {
+  const major = Number(process.version.match(/v([^.]+)/)[1])
+  if (major < minNodeVersion) {
+    throw Error(`yargs supports a minimum Node.js version of ${minNodeVersion}. Read our version support policy: https://github.com/yargs/yargs#supported-nodejs-versions`)
```

Comment:
I've started throwing on unsupported versions of Node.js, otherwise folks can get bit somewhere down the road, e.g., when we pull in a feature like async generators, which aren't supported in Node 8.